Source/Core/Core/PowerPC/JitArm64/JitArm64_LoadStore.cpp

```diff
@@ -957,11 +973,40 @@ void JitArm64::dcbt(UGeckoInstruction inst)
   // This is important because invalidating the block cache when we don't
   // need to is terrible for performance.
   // (Invalidating the jit block cache on dcbst is a heuristic.)
-  if (CanMergeNextInstructions(1) && js.op[1].inst.OPCD == 31 && js.op[1].inst.SUBOP10 == 54 &&
-      js.op[1].inst.RA == inst.RA && js.op[1].inst.RB == inst.RB)
+  const bool followed_by_dcbst =
+      (CanMergeNextInstructions(1) && js.op[1].inst.OPCD == 31 && js.op[1].inst.SUBOP10 == 54 &&
+       js.op[1].inst.RA == inst.RA && js.op[1].inst.RB == inst.RB);
+
+  if (followed_by_dcbst)
   {
     js.skipInstructions = 1;
   }
+
+  const u32 i = followed_by_dcbst;
+  // Check if the next instructions match a known looping pattern:
+  // - dcbt rX
+  // - dcbst rX (optional)
+  // - addi rX,rX,32
+  // - bdnz+ -8 (or -12 if dcbst)
+  const bool make_loop = inst.RA == 0 && inst.RB != 0 && CanMergeNextInstructions(2 + i) &&
+                         (js.op[1 + i].inst.hex & 0xfc00'ffff) == 0x38000020 &&
+                         js.op[1 + i].inst.RA_6 == inst.RB && js.op[1 + i].inst.RD_2 == inst.RB &&
+                         js.op[2 + i].inst.hex == 0x42010000 - (4 * (2 + i));
+
+  if (make_loop)
+  {
+    u32 cycle_count_per_loop = 0;
+    for (u32 j = 0; j < 3 + i; j++)
+      cycle_count_per_loop += js.op[j].opinfo->num_cycles;
+    const ARM64Reg WA = gpr.GetScopedReg();
```

Comment:
Converting the `GetScopedReg` result from `ScopedARM64Reg` to `ARM64Reg` like this means the register immediately gets unlocked in the register cache. This is unsafe. You need to keep the result of `GetScopedReg` as `ScopedARM64Reg`, which is why the existing calls to `GetScopedReg` use `auto`.